sys/kern/vfs_syscalls.c

```diff
@@ -406,6 +406,9 @@ int do_chdir(proc_t *p, const char *path) {
   if ((error = vfs_namelookup(path, &cwd)))
     return error;
 
+  if (cwd->v_type != V_DIR)
+    return ENOTDIR;
```

Comment:
Something is missing here. Please carefully read `vfs_namelookup` docstring ;-)

Reply:
Thanks! I added dropping the reference to this vnode